### DISCUSSION TITLE
Пользовательские заметки к любым парам

### DISCUSSION THREAD

**Discussion by lSpaceCatl** [0niel/university-app#219] [Идеи]
**Заметки**

Необходимо создать поле «заметок» для каждой из парю куда каждый студент пишет произвольный текст. Заметка автоудалится после того, как пара кончилась

Пример реализации:
Сделать поле с названием пары/аудиторией активной кнопкой. При нажатии появляется расширенная информация: на каких неделях эта пара, преподаватель и  поле для ввода произвольного текста. Произвольный текст отображается только на той паре, где он был введен (такие же пары на следующих неделях не должны содержать данного текста)
Заметка должна отображаться в САМОМ РАСПИСАНИИ. Ввод заметки или ее изменение - через активную кнопку пары. 

Для чего?
Например преподаватель сообщает что через неделю на паре будет контроха
Функция «заметок» позволит записать это событие и при дальнейшем просмотре расписания даст подготовиться

**Comment by MadL1me** [comment]
Мне кажется, вместо заметок, лучше сделать пользовательские "лейблы" (ака лейблы в Trello/Jira/Github и тд). В чем преимущество? Ты можешь задефайнить несколько штук и реюзать их, и при том что лейблы можно сильно ограничевать по размеру, лейблы можно будет рендерить на экране пар. Вообще, идею с лейблами можно переквалифицировать под отдельную ишью btw.

**Comment by ValeryVerkhoturov** [comment]
> Мне кажется, вместо заметок, лучше сделать пользовательские "лейблы" (ака лейблы в Trello/Jira/Github и тд). В чем преимущество? Ты можешь задефайнить несколько штук и реюзать их, и при том что лейблы можно сильно ограничевать по размеру, лейблы можно будет рендерить на экране пар. Вообще, идею с лейблами можно переквалифицировать под отдельную ишью btw.

Нравится. Но встает выбор между лейблами и автообновлением #57

**Comment by MadL1me** [comment]
> Нравится. Но встает выбор между лейблами и автообновлением #57  

При заметках будет таже самая проблема. И при этом оно не то что сложно решается при автообновлении код просто будет чекать, есть ли лейблы на паре такого же типа, и все, и оставлять их в случае чего.

**Comment by ValeryVerkhoturov** [comment]
> При заметках будет таже самая проблема. И при этом оно не то что сложно решается при автообновлении код просто будет чекать, есть ли лейблы на паре такого же типа, и все, и оставлять их в случае чего.

Заметка может быть прикреплена ко дню. Лейбл нужен предмету. Хорошо, если расписание в локальном хранилище и на сервере совпадут. А если нет? Есть несколько решений:
1. удалять без ведома пользователя (будут контрольные - сюрпризы, просроченные дедлайны)
2. оставлять без изменений (лейблы будут пропадать с нужных предметов, появляться у ненужных)
3. спрашивать, оставлять лейбл или нет (если будет много лейблов, то отнимет много времени просмотреть, причем в неподходящий момент)
4. отказываться от автообновления. Тогда если пользователь захочет обновиться:
- уведомлять о пункте 1 (разумно, но неприятно)
- уведомлять, что обновление может потратить время пользователя и использовать пункт 3

**Comment by MadL1me** [comment]
@lSpaceCatl @ValeryVerkhoturov завел отдельную ишью: #75  по тегам, мне кажется будет круто если у нас будут и теги и заметки - у каждой штуки есть свои сверх полезные юзкейсы. На выходных займусь и тем и другим, если получится, начну с заметок

**Comment by Dragonprod** [comment]
Дубликат #213
